query/stdlib/influxdata/influxdb/rules_test.go

```diff
@@ -118,3 +121,216 @@ func TestPushDownRangeRule(t *testing.T) {
+			ReadRangePhysSpec: influxdb.ReadRangePhysSpec{
+				Bucket: "my-bucket",
+				Bounds: flux.Bounds{
+					Start: fluxTime(5),
+					Stop:  fluxTime(10),
+				},
+			},
+			ValueColumnName: execute.DefaultValueColLabel,
+		}
+		if filter {
+			s.FilterSet = true
+			s.Filter = filterSpec.Fn
+		}
+		return &s
+	}
+
+	tests := []plantest.RuleTestCase{
+		{
+			Name: "simple",
+			// from -> range -> keys -> keep -> distinct  =>  ReadTagKeys
+			Rules: []plan.Rule{
+				influxdb.PushDownRangeRule{},
+				influxdb.ReadTagKeysRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("from", &fromSpec),
+					plan.CreateLogicalNode("range", &rangeSpec),
+					plan.CreateLogicalNode("keys", &keysSpec),
+					plan.CreateLogicalNode("keep", &keepSpec),
+					plan.CreateLogicalNode("distinct", &distinctSpec),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{1, 2},
+					{2, 3},
+					{3, 4},
+				},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreatePhysicalNode("ReadRange", readTagKeysSpec(false)),
+				},
+			},
+		},
+		{
+			Name: "with filter",
+			// from -> range -> filter -> keys -> keep -> distinct  =>  ReadTagKeys
+			Rules: []plan.Rule{
+				influxdb.PushDownRangeRule{},
+				influxdb.PushDownFilterRule{},
+				influxdb.ReadTagKeysRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("from", &fromSpec),
+					plan.CreateLogicalNode("range", &rangeSpec),
+					plan.CreateLogicalNode("filter", &filterSpec),
+					plan.CreateLogicalNode("keys", &keysSpec),
+					plan.CreateLogicalNode("keep", &keepSpec),
+					plan.CreateLogicalNode("distinct", &distinctSpec),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{1, 2},
+					{2, 3},
+					{3, 4},
+					{4, 5},
+				},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreatePhysicalNode("merged_ReadRange_filter", readTagKeysSpec(true)),
+				},
+			},
+		},
+		{
+			Name: "with successor",
+			// from -> range -> keys -> keep -> distinct -> count  =>  ReadRange -> count
+			Rules: []plan.Rule{
+				influxdb.PushDownRangeRule{},
+				influxdb.ReadTagKeysRule{},
+			},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("from", &fromSpec),
+					plan.CreateLogicalNode("range", &rangeSpec),
+					plan.CreateLogicalNode("keys", &keysSpec),
+					plan.CreateLogicalNode("keep", &keepSpec),
+					plan.CreateLogicalNode("distinct", &distinctSpec),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{1, 2},
+					{2, 3},
+					{3, 4},
+					{4, 5},
+				},
+			},
+			After: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreatePhysicalNode("ReadRange", readTagKeysSpec(false)),
+					plan.CreatePhysicalNode("count", &universe.CountProcedureSpec{}),
+				},
+				Edges: [][2]int{{0, 1}},
+			},
+		},
+		{
+			Name: "with multiple successors",
```

Comment:
I think this is a `ReadRange` test. Perhaps you meant to write a similar test for `ReadTagKeys`? I'm fine if you just want to remove it.

Reply:
I intentionally copied this. I thought it would be a useful test for here too.

Reply:
Updated the comments.